tools/Microsoft.AspNet.Server.Kestrel.GeneratedCode/FrameFeatureCollection.cs

```diff
@@ -25,26 +25,40 @@ public virtual void BeforeCompile(BeforeCompileContext context)
 
         public static string GeneratedFile()
         {
-            var commonFeatures = new[]
+            var alwaysFeatures = new[]
             {
                 typeof(IHttpRequestFeature),
                 typeof(IHttpResponseFeature),
                 typeof(IHttpRequestIdentifierFeature),
-                typeof(IHttpSendFileFeature),
                 typeof(IServiceProvidersFeature),
-                typeof(IHttpAuthenticationFeature),
                 typeof(IHttpRequestLifetimeFeature),
+                typeof(IHttpConnectionFeature)
+            };
+
+            var commonFeatures = new[]
+            {
+                typeof(IHttpAuthenticationFeature),
                 typeof(IQueryFeature),
-                typeof(IFormFeature),
+                typeof(IFormFeature)
+            };
+
+            var sometimesFeatures = new[]
+            {
+                typeof(IHttpUpgradeFeature),
```

Comment:
Only does always+common+implemented. ```IHttpUpgradeFeature``` gets picked up because its in the implemented group as part of frame so needs resetting back to ```this```